File: white_backend.jl
module Backend

export contacts, chats, update_contact_list, get_key, set_key, get_state, set_state

contacts = Dict()
chats = Dict()

function update_contact_list(message)
    id = message["message"]["from"]["id"]
    if !(id in  keys(contacts))
        contacts[id] = Dict("state"=> 0, "message"=>message)
    end
end

function get_key(id, key)
    return contacts[id][key]
end

function set_key(id, key, value)
    contacts[id][key] = value
end

function get_state(id)
    return get_key(id,"state")
end

function set_state(id, state)
    set_key(id,"state", state)
end

end

File: white_backend.py
import logging
log = logging.getLogger("pizza_bot")

contacts = {}
chats = {}

def update_contact_list(message):
    id = message["message"]["from"]["id"]
    if id not in  contacts.keys():
        contacts[id] = {"state": 0, "message":message}
    
def get_key(id, key):
    return contacts[id][key]

def set_key(id, key, value):
    contacts[id][key] = value

def get_state(id):
    return get_key(id,"state")

def set_state(id, state):
    set_key(id,"state", state)



